custom_components/tuya_local/devices/friedrich_airconditioner.yaml

```diff
@@ -0,0 +1,136 @@
+name: Air Conditioner
+products:
+  - id: vfdma8hzkgpdkwlm
+    # https://www.friedrich.com/professional/products/uni-fit
+    # NB: Tuya calls this "Friedrich Window Unit" but these are
+    # through-the-wall units
+    name: Friedrich Uni-Fit Through-the-Wall AC (Models UCT14A30, UCT12A30)
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: cold
+              value: cool
+            - dps_val: wet
+              value: dry
+            - dps_val: wind
+              value: fan_only
+            - dps_val: auto
+              value: auto
+    - id: 2
+      type: integer
+      name: temperature
+      unit: F
+      range:
+        min: 69
+        max: 89
+    - id: 3
+      type: integer
+      name: current_temperature
+    - id: 4
+      type: string
+      name: mode
+      hidden: true
+    - id: 5
+      type: integer
+      name: fan_mode
+      mapping:
+        - dps_val: 3
+          value: low
+        - dps_val: 2
+          value: medium
+        - dps_val: 1
+          value: high
+        - dps_val: 4
+          value: auto
+    - id: 8
+      type: boolean
+      optional: true
+      name: preset_mode
+      mapping:
+        - dps_val: true
+          value: eco
+        - dps_val: false
+          value: comfort
+        - dps_val: null
+          value: comfort
+          hidden: true
+secondary_entities:
+  - entity: switch
+    name: "Sleep Mode"
+    icon: "mdi:sleep"
+    translation_key: sleep
+    category: config
+    dps:
+      - id: 101
+        type: boolean
+        name: switch
+  - entity: sensor
+    # false is C, true is F
+    name: "Configured Temperature Unit"
```

Comment:
This is not very useful here as an informational sensor. If the value is  C then the climate entity will not be working properly, as the range and unit there is hard coded to F. If it is read-only, then it would be better just including it as the temperature_unit dp on the climate entity, and setting the range conditionally.